src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -1445,7 +1479,7 @@ bool AbstractWriteLog<I>::check_allocation(C_BlockIORequestT *req,
     /* Don't attempt buffer allocate if we've exceeded the "full" threshold */
     if (m_bytes_allocated + bytes_allocated > bytes_allocated_cap) {
       if (!req->has_io_waited_for_buffers()) {
-        req->set_io_waited_for_entries(true);
+        req->set_io_waited_for_buffers(true);
```

Comment:
@majianpeng  can you help to check if the `set_io_waited_for_entries` is useless anywhere include line 1471 and review the `set_io_waited_for_buffers` logic?

Reply:
I think this is ok. But  i think we need split this patch into two patch. one for split larger extent. Left for other patch which                  fix the bug which change waited_for_entries to waited_for_buffers.